tests/acceptance/common.sh

```diff
@@ -0,0 +1,120 @@
+function log() {
+    local level="${2:-INFO}"
+    echo "[$(TZ=UTC date --rfc-3339=ns) ${level} ${BATS_TEST_NAME}] $1"
+}
+
+function logError() {
+    log $1 'ERROR'
+}
+
+function logWarn() {
+    log $1 'WARN'
+}
+
+function testAzureSecret() {
+    local role_name="$1"
+    local subscription_id="$2"
+    local resource_group_name="$3"
+    local vault_role_name="$4"
+    local config_dir="$5"
+    local engine_name="$6"
+    local ttl=5
+
+    log "Creating Azure secret, azure_role='${role_name}', vault_role='${vault_role_name}'"
+    vault write "${engine_name}/roles/${vault_role_name}" ttl="${ttl}" azure_roles=-<<EOF
+[
+    {
+        "role_name": "${role_name}",
+        "scope":  "/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}"
+    }
+]
+EOF
+
+    local secret_file="${config_dir}/secret-${vault_role_name}.json"
+    vault read "${engine_name}/creds/${vault_role_name}" -format=json > ${secret_file}
+
+    sp_id=$(cat "${secret_file}" | jq -er .data.client_id)
+    log "Secret created successfully, azure_role='${role_name}', vault_role='${vault_role_name}', file=${secret_file}"
+
+    local found=''
+    for n in {0..30}
+    do
+         if ! az ad sp show --id "${sp_id}" > /dev/null ; then
```

Comment:
Is it possible to use bats primitives in this function? Like `run` and then checking `$status`? Just thinking it might be easier to read and would take care of the output without redirecting to /dev/null.

Reply:
I am not sure `run` fits that well in this context. I could perhaps factor this portion out to its own function which may help with the readability issue that you mention.

Reply:
Fixed in https://github.com/hashicorp/vault-plugin-secrets-azure/pull/85/commits/322a5da4372af9d3a6295372ed241f6ddebcb660